### PR TITLE
[New Rule] Azure Conditional Access Policy Modified

### LINKED ISSUE #143: [New Rule] Azure Conditional Access Policy Modified
## Description

Azure Conditional Access policies control access to resources via if-then statements. For example, if a user wants to access a resource, then they must complete an action such as using multi-factor authentication to access it. An adversary may modify a Conditional Access policy in order to weaken their target's security controls.

```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy"
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Target Operating Systems:**

- **Platforms**

Azure

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs** None

## Optional Info
- **References:**
  - https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview

## Example Data

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #143 

## Summary
Identifies when an Azure Conditional Access policy is modified. Azure Conditional Access policies control access to resources via if-then statements. For example, if a user wants to access a resource, then they must complete an action such as using multi-factor authentication to access it. An adversary may modify a Conditional Access policy in order to weaken their target's security controls.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[metadata]
+creation_date = "2020/09/01"
+ecs_version = ["1.6.0"]
+maturity = "development"
```
Comment (brokensound77) [inline rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml:4]:
```suggestion
maturity = "production"
```
?

**Review by Samirbous — APPROVED** [review]

File: rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml (hunk 1 above)
Comment (threat-punter) [reply rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml:4]: Good catch. Thanks

**Comment by threat-punter** [comment]
> Could we add a technique like `Account Manipulation` to this? `event.outcome` didn't appear to populate when I looked at the results during our test, not sure if that is everytime though. What about adding `azure.activitylogs.properties.result:success`?
> 
> update - event outcome doesn't populate with all activitylogs

I added the ATT&CK metadata to the rule. Let me check on the ECS mapping before I include that non-ECS field in the rule. Thanks

File: rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+references = ["https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview"]
+risk_score = 47
+rule_id = "bc48bba7-4a23-4232-b551-eca3ca1e3f20"
+severity = "medium"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy"
+'''
```
Comment (threat-punter) [inline rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml:30]:
```suggestion
'''


[[rule.threat]]
framework = "MITRE ATT&CK"
[[rule.threat.technique]]
id = "T1098"
name = "Account Manipulation"
reference = "https://attack.mitre.org/techniques/T1098/"


[rule.threat.tactic]
id = "TA0003"
name = "Persistence"
reference = "https://attack.mitre.org/tactics/TA0003/"

```

**Review by brokensound77 — APPROVED** [review]

**Review by devonakerr — APPROVED** [review]

**Comment by threat-punter** [comment]
Confirmed updated query matches expected events

image

**Comment by threat-punter** [comment]
See Beats issue https://github.com/elastic/beats/issues/20990 re an update that's required for us to use `event.outcome` in this rule.

Confirmed that the updated query matches the expected events, but the tests are failing when I add `and azure.activitylogs.properties.result:success` to the query. 🤔

```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy" and azure.activitylogs.properties.result:success
```

image

File: rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+note = "The Azure Filebeat module must be enabled to use this rule."
+references = ["https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview"]
+risk_score = 47
+rule_id = "bc48bba7-4a23-4232-b551-eca3ca1e3f20"
+severity = "medium"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy" and azure.activitylogs.properties.result:success
```
Comment (threat-punter) [inline rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml:29]:
```suggestion
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy" and event.outcome:success
```

**Comment by threat-punter** [comment]
This rule can be merged when https://github.com/elastic/beats/issues/20990 is resolved.

**Comment by threat-punter** [comment]
PR https://github.com/elastic/beats/pull/20998 has been merged to ensure that field `azure.activitylogs.properties.result` is renamed to `event.outcome`. This PR can be merged. When a Filebeat 7.10 BC is available, I'll make this it's working as expected.

**Comment by bm11100** [comment]
We should update this to account for the AuditLogs as well
```
event.module:azure and (event.dataset:azure.activitylogs or event.dataset:azure.auditlogs) and (azure.activitylogs.operation_name:"Update policy" or azure.auditlogs.operation_name:"Update policy")
```

File: rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+note = "The Azure Filebeat module must be enabled to use this rule."
+references = ["https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview"]
+risk_score = 47
+rule_id = "bc48bba7-4a23-4232-b551-eca3ca1e3f20"
+severity = "medium"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Update policy" and event.outcome:success
```
Comment (threat-punter) [inline rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml:29]: I'm updating this query, as this event appears in both the Azure audit logs and activity logs (good find @bm11100).

```suggestion
event.module:azure and (event.dataset:azure.activitylogs or event.dataset:azure.auditlogs) and (azure.activitylogs.operation_name:"Update policy" or azure.auditlogs.operation_name:"Update policy") and event.outcome:success
```

image

Comment (bm11100) [reply rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml]: I dont think the Audit Logs use `event.outcome` so we could do the below
```
event.module:azure and event.dataset:(azure.activitylogs or azure.auditlogs) and (azure.activitylogs.operation_name:"Update policy" or azure.auditlogs.operation_name:"Update policy")
```

Comment (bm11100) [reply rules/azure/defense_evasion_azure_conditional_access_policy_modified.toml]: Actually i see the comment above, this was fixed.

**Comment by threat-punter** [comment]
I took the liberty of tweaking rule, `Possible Consent Grant Attack via Azure-Registered Application` in this PR too. This rule was just merged. I amended the query so that it searches both the Azure audit logs and activity logs, as this event appears in both of those logs.

I'll request a review of this PR again, but I did confirm that both queries in this PR match the intended events.

**Review by bm11100 — APPROVED** [review]
lgtm, with the changes going into 7.10 the activity logs will eval true as well

File: rules/azure/initial_access_consent_grant_attack_via_azure_registered_application.toml [hunk 5]
```diff
@@ -31,7 +31,12 @@ tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Identity and Acc
 type = "query"
 
 query = '''
-event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Consent to application" and event.outcome:success
+event.module:azure and event.dataset:(azure.activitylogs or azure.auditlogs) and
+  (
+    azure.activitylogs.operation_name:"Consent to application" or 
+    azure.auditlogs.operation_name:"Consent to application"
+  )
+  and event.outcome:success
```
Comment (threat-punter) [inline rules/azure/initial_access_consent_grant_attack_via_azure_registered_application.toml:39]:
```suggestion
  ) and
  event.outcome:success
```

**Review by bm11100 — COMMENTED** [review]
🔥

**Review by devonakerr — APPROVED** [review]

**Pull request merged**